src/cc/bcc_perf_map.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2016 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "bcc_perf_map.h"
+
+int bcc_perf_map_nspid(int pid) {
+  char status_path[64];
+  FILE *status;
+
+  snprintf(status_path, sizeof(status_path), "/proc/%d/status", pid);
+  status = fopen(status_path, "r");
+
+  if (!status)
+    return -1;
+
+  // return the original PID if the NSpid line is missing
+  int nspid = pid;
+
+  size_t size;
+  char *line = NULL;
+  while (getline(&line, &size, status) != -1)
```

Comment:
Thanks for the feedback!

Fixing `size` is fine. My `getline` man page actually says it's ignored if `*lineptr == NULL`, but http://man7.org/linux/man-pages/man3/getline.3.html says it must be zero. Regardless, there's clearly there's no downside to intializing to zero so I'll do that.

Re: `getline` allocating on every call: are you sure about that? http://man7.org/linux/man-pages/man3/getline.3.html says `getline` will use the buffer passed and `realloc` it if necessary. It doesn't say it'll allocate every time around and the example code in the man page just has a single `free` outside the loop. Additionally, my own simple test with valgrind appears to confirm this is safe:

https://gist.github.com/markdrayton/1ef00109ff2092898ab3b908d0598c41